staging/src/k8s.io/client-go/tools/cache/shared_informer_test.go

```diff
@@ -196,59 +215,60 @@ func TestResyncCheckPeriod(t *testing.T) {
 
 	// listener 1, never resync
 	listener1 := newTestListener("listener1", 0)
-	informer.AddEventHandlerWithResyncPeriod(listener1, listener1.resyncPeriod)
+	listener1Registration, _ := informer.AddEventHandlerWithResyncPeriod(listener1, listener1.resyncPeriod)
+
 	if e, a := 12*time.Hour, informer.resyncCheckPeriod; e != a {
 		t.Errorf("expected %d, got %d", e, a)
 	}
-	if e, a := time.Duration(0), informer.processor.listeners[0].resyncPeriod; e != a {
+	if e, a := time.Duration(0), informer.processor.getListener(listener1Registration).resyncPeriod; e != a {
```

Comment:
optional nit: for readability I'd consider declaring `gl := informer.processor.getListener` so that the `informer.processor.getListener(listener1Registration)` lines can be much shorter